lib/runtime/wazero/imports.go

```diff
@@ -2188,8 +2357,33 @@ func ext_storage_root_version_1(ctx context.Context, m api.Module) uint64 {
 }
 
 func ext_storage_root_version_2(ctx context.Context, m api.Module, version uint32) uint64 { //skipcq: RVV-B0012
-	// TODO: update to use state trie version 1 (#2418)
-	return ext_storage_root_version_1(ctx, m)
+	rtCtx := ctx.Value(runtimeContextKey).(*runtime.Context)
+	if rtCtx == nil {
+		panic("nil runtime context")
+	}
+	storage := rtCtx.Storage
+
+	stateVersionBytes, _ := m.Memory().Read(version, 4)
+	_, err := trie.ParseVersion(binary.LittleEndian.Uint32(stateVersionBytes))
+	if err != nil {
+		logger.Errorf("failed parsing state version: %s", err)
+		return 0
```

Comment:
Since `ext_storage_root_version_2` caller expects a `Vec<u8>` I would suggest that in failure cases return a pointer to an empty `Vec<u8>` instead of 0

https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/primitives/io/src/lib.rs#L316C2-L316C2

Reply:
Yes, I would suggest using the `mustWrite` function so you don't need to worry about errors and panics, and you can define a variable like `emptyByteVectorEncoded` and use it everywhere you need 

```go
// on top of imports.go

var (
 ...
 emptyByteVectorEncoded []byte = scale.MustMarshal([]byte{})
)

// in the ext_storage_root_version_2 function 
func ext_storage_root_version_2(...) ... {
   ...
   if err != nil {
       return mustWrite(m, rtCtx.Allocator, emptyByteVectorEncoded)
   }
   ...
}


```

Reply:
Great! thanks!